avocado/utils/lv_utils.py

```diff
@@ -0,0 +1,517 @@
+    vg_ramdisk_dir = os.path.join(ramdisk_basedir, vg_name)
+    ramdisk_filename = os.path.join(vg_ramdisk_dir,
+                                    ramdisk_sparse_filename)
+    vg_ramdisk_cleanup(ramdisk_filename, vg_ramdisk_dir, vg_name)
+    result = ""
+    if not os.path.exists(vg_ramdisk_dir):
+        os.mkdir(vg_ramdisk_dir)
+    try:
+        LOGGER.debug("Mounting tmpfs")
+        result = process.run("mount -t tmpfs tmpfs %s" %
+                             vg_ramdisk_dir, shell=True)
+
+        LOGGER.debug("Converting and copying /dev/zero")
+        if disk:
+            vg_size = get_diskspace(disk)
+        cmd = ("dd if=/dev/zero of=%s bs=1M count=1 seek=%s" %
+               (ramdisk_filename, vg_size))
+        result = process.run(cmd, shell=True)
+        if not disk:
+            LOGGER.debug("Finding free loop device")
+            result = process.run("losetup --find", shell=True)
+    except process.CmdError, ex:
+        LOGGER.error(ex)
+        vg_ramdisk_cleanup(ramdisk_filename, vg_ramdisk_dir, vg_name)
+        raise ex
+
+    if not disk:
+        loop_device = result.stdout.rstrip()
+    else:
+        loop_device = disk
+    try:
+        if not disk:
+            LOGGER.debug("Creating loop device")
+            result = process.run("losetup %s %s" %
+                                 (loop_device, ramdisk_filename), shell=True)
+        LOGGER.debug("Creating physical volume %s", loop_device)
+        result = process.run("pvcreate %s" % loop_device, shell=True)
+        LOGGER.debug("Creating volume group %s", vg_name)
+        result = process.run("vgcreate %s %s" %
+                             (vg_name, loop_device), shell=True)
+    except process.CmdError, ex:
+        LOGGER.error(ex)
+        vg_ramdisk_cleanup(ramdisk_filename, vg_ramdisk_dir,
+                           vg_name, loop_device)
+        raise ex
+
+    LOGGER.debug(result.stdout.rstrip())
+
+
+def vg_ramdisk_cleanup(ramdisk_filename=None, vg_ramdisk_dir=None,
+                       vg_name=None, loop_device=None):
+    """
+    Inline cleanup function in case of test error.
+    """
+    if vg_name is not None:
+        loop_device = re.search(r"([/w]+) %s lvm2" % vg_name,
+                                process.run("pvs").stdout)
+        if loop_device is not None:
+            loop_device = loop_device.group(1)
+        result = process.run("vgremove %s" %
+                             vg_name, ignore_status=True, shell=True)
+        if result.exit_status == 0:
+            LOGGER.debug(result.stdout.rstrip())
+        else:
+            LOGGER.debug("%s -> %s", result.command, result.stderr)
+
+    if loop_device is not None:
+        result = process.run("pvremove %s" % loop_device,
+                             ignore_status=True, shell=True)
+        if result.exit_status == 0:
+            LOGGER.debug(result.stdout.rstrip())
+        else:
+            LOGGER.debug("%s -> %s", result.command, result.stderr)
```

Comment:
similar story as with vg, only this time non-zero return should be failure as we only call this with the loop_device when it's already created. So I'd just use:

```
if result.exit_status != 0:
    LOGGER.error("Failed to wipe pv from %s", loop_device)
```
(either error or warn)